scrape/target_test.go

```diff
@@ -508,19 +508,30 @@ func TestBucketLimitAppender(t *testing.T) {
 	}
 
 	cases := []struct {
-		h           histogram.Histogram
-		limit       int
-		expectError bool
+		h                 histogram.Histogram
+		limit             int
+		expectError       bool
+		expectBucketCount int
+		expectSchema      int32
 	}{
 		{
 			h:           example,
 			limit:       3,
 			expectError: true,
 		},
 		{
-			h:           example,
-			limit:       10,
-			expectError: false,
+			h:                 example,
+			limit:             4,
+			expectError:       false,
+			expectBucketCount: 4,
+			expectSchema:      -1,
+		},
+		{
+			h:                 example,
```

Comment:
please add my bigGap test case to cover that

Reply:
Done